azure/scope/cluster.go

```diff
@@ -1020,9 +1020,18 @@ func (s *ClusterScope) SetControlPlaneSecurityRules() {
 	if !s.ControlPlaneEnabled() {
 		return
 	}
-	if s.ControlPlaneSubnet().SecurityGroup.SecurityRules == nil {
+
+	subnet := s.ControlPlaneSubnet()
```

Comment:
in this flow where are we adding any user-provided security rules? (for example if a user specifies TCP `9345`)

or is that elsewhere and the purpose of this change is to filter out `22` and apiserver port if it's not included?

Reply:
This flow performs only defaulting on top of the user-provided set of rules, which may not be empty. This way having a field populated does not always need to specify all allowed ports, only additional ones.

I’m currently validating how this works in tandem with ClusterClass definitions we have. The desired goal is to permit specifying `registrationMethod: control-plane-endpoint` and allow RKE2 to register on the allowed port. But ideally this should follow internal name resolution, so that the traffic wouldn’t go through external load balancer.

If it doesn’t work, maybe it would still require additional LB rules.

Reply:
Sorry, maybe I’m not fully understanding. I will try to give my thoughts on this:

I was attempting to describe the flow of how the rules are processed. We only apply defaults on top of what the user provides, so the user can still specify ports like TCP 9345 explicitly in security rules. We’re not overriding or stripping anything out, just adding fallback values when needed. LB rules were needed as well, since CP endpoint is external from the cluster perspective.

Current set of changes is what was required to make cluster healthy, and added e2e test displays that. My previous comment was a set of assumptions.